lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -195,6 +200,31 @@ import qualified Data.ByteString.Lazy as BL
 import qualified Data.Text as T
 import qualified Data.Text.Encoding as T
 
+{-------------------------------------------------------------------------------
+                               Style of Wallets
+-------------------------------------------------------------------------------}
+
+data WalletStyle
+    = Shelley
+    | Random
+    | Icarus
+    | Trezor
+    | Ledger
+
+type family WalletStyleSymbol (style :: WalletStyle) :: Symbol where
+    WalletStyleSymbol 'Shelley = "shelley"
+    WalletStyleSymbol 'Random  = "random"
+    WalletStyleSymbol 'Icarus  = "icarus"
+    WalletStyleSymbol 'Trezor  = "trezor"
+    WalletStyleSymbol 'Ledger  = "ledger"
+
+type family AllowedMnemonics (style :: WalletStyle) :: [Nat] where
+    AllowedMnemonics 'Shelley = '[15,18,21,24]
+    AllowedMnemonics 'Random  = '[12]
+    AllowedMnemonics 'Icarus  = '[15]
+    AllowedMnemonics 'Trezor  = '[12,15,18,21,24]
+    AllowedMnemonics 'Ledger  = '[12,15,18,21,24]
```

Comment:
> perhaps we should adjust our coding standard document?

I think we have found a few cases where vertical alignment can make sense, especially when it's not exaggerated. For things like, function signatures, record declarations and most case pattern-match newlines are still the de-facto approach IMO. 

I am not sure whether we should adjust the coding standard however. These are coding standards and, as developers (and reviewer), we should know and tolerate when it is reasonable to deviate. Perhaps we could give some extra examples of good usage of vertical alignment (explaining also _why_ the alignment makes sense in that case) and, be more clear about the intent of the coding standards: they are guidelines, not hard constraints.

Reply:
I do think some extra clarity in the coding standard would be helpful.

The main reason is that the standard specifically marks this kind of alignment as "BAD".

I couldn't find an example that's exactly the same as the code in your PR, but this example is typical of that given in the standard (with `::` instead of `=`):

```hs
-- BAD
data MyRecord = MyRecord 
    { _myRecordLongNameField :: !String
    , _myRecordShort         :: ![Int]
    }
```

Now I don't think the above is necessarily bad (hence my original comment), but since we are seeing a pattern (whereby people are using this kind of alignment more regularly), wouldn't it be better to be explicit about what we do and don't want, and then update the standard accordingly?